src/array.ts

```diff
@@ -113,9 +123,14 @@ export function of() {
  * @return The filled target
  */
 export function fill<T>(target: ArrayLike<T>, value: any, start?: number, end?: number): ArrayLike<T> {
+	if (has('array-fill')) {
+		const method = (<any> Array.prototype).fill;
+		return method.call.apply(method, arguments);
+	}
+
 	const length = toLength(target.length);
 	let i = normalizeOffset(toInteger(start), length);
-	end = normalizeOffset(end ? toInteger(end) : length, length);
+	end = normalizeOffset(end === undefined ? length : toInteger(end), length);
```

Comment:
The spec only defaults to length when end is undefined and uses toInteger() for all other values: http://www.ecma-international.org/ecma-262/6.0/#sec-array.prototype.fill